geomesa-spark/geomesa-spark-sql/src/main/scala/org/locationtech/geomesa/spark/sedona/package.scala

```diff
@@ -0,0 +1,49 @@
+/***********************************************************************
+ * Copyright (c) 2013-2021 Commonwealth Computer Research, Inc.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Apache License, Version 2.0
+ * which accompanies this distribution and is available at
+ * http://www.opensource.org/licenses/apache2.0.php.
+ ***********************************************************************/
+
+package org.locationtech.geomesa.spark
+
+import org.apache.sedona.sql.UDF.Catalog
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.sedona_sql.strategy.join.JoinQueryDetector
+
+package object sedona {
+
+  // User can specify a common prefix of UDFs/UDAFs introduced by Apache Sedona. For example, when prefix is specified
+  // as "Sedona_", ST_Contains function from Apache Sedona will be named as "Sedona_ST_Contains". When prefix is explicitly
+  // set to empty, Apache Sedona functions will replace corresponding Spark JTS functions.
+  def sedonaUdfPrefix(sqlContext: SQLContext): String = sqlContext.getConf("spark.geomesa.sedona.udf.prefix", "sedona_")
+
+  /**
+   * Register Geometry UDTs, UDFs, UDAFs and optimization rules for Apache Sedona.
+   * @param sqlContext Spark [[SQLContext]] object
+   */
+  def initSedona(sqlContext: SQLContext): Unit = {
+    val prefix = sedonaUdfPrefix(sqlContext)
+    registerOptimizations(sqlContext)
+    registerUdfs(sqlContext, prefix)
+  }
+
+  private def registerOptimizations(sqlContext: SQLContext): Unit = {
+    Seq(SedonaGeometryLiteralRules).foreach { r =>
+      if (!sqlContext.experimental.extraOptimizations.contains(r))
+        sqlContext.experimental.extraOptimizations ++= Seq(r)
+    }
+    Seq(new JoinQueryDetector(sqlContext.sparkSession)).foreach { s =>
+      if(!sqlContext.experimental.extraStrategies.contains(s))
+        sqlContext.experimental.extraStrategies ++= Seq(s)
+    }
+  }
+
+  private def registerUdfs(sqlContext: SQLContext, prefix: String): Unit = {
+    val sparkSession = sqlContext.sparkSession
+    Catalog.expressions.foreach(f => sparkSession.sessionState.functionRegistry.createOrReplaceTempFunction(
+      prefix + f.getClass.getSimpleName.dropRight(1), f))
+    Catalog.aggregateExpressions_UDAF.foreach(f => sparkSession.udf.register(prefix + f.getClass.getSimpleName, f))
```

Comment:
Does this also include the sedona UDAFs? https://github.com/apache/incubator-sedona/blob/5055078d18c687f92fcc67b7377203c501823b8a/sql/src/main/scala/org/apache/sedona/sql/UDF/UdfRegistrator.scala#L44

Reply:
yes:

```
spark.sessionState.functionRegistry.listFunction.map(_.funcName).sorted.filter(_.toUpperCase.contains("AGGR")).foreach(println)
sedona_st_envelope_aggr
sedona_st_intersection_aggr
sedona_st_union_aggr

```